### PR TITLE
Add a Maui Class Library template

### PR DESCRIPTION
**Pull request by mattleibow**
### Description of Change ###

This PR adds the support for a class library:

```
dotnet new mauilib
```

### CODE REVIEW

File: src/Templates/src/templates/maui-lib/Platforms/iOS/PlatformClass1.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿using System;
+
+namespace MauiLib1
+{
+	// All the code in this file is only included on iOS.
```
Comment (Eilon) [inline src/Templates/src/templates/maui-lib/Platforms/iOS/PlatformClass1.cs:5]: Doesn't this also get used on MacCatalyst?

File: src/Templates/src/templates/maui-lib/Class1.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+
+namespace MauiLib1
+{
+	// All the code in this file is shared between all the platforms
+	// including net6.0 shared library.
```
Comment (Eilon) [inline src/Templates/src/templates/maui-lib/Class1.cs:6]: The phrasing here doesn't ring well to me. I see that you're trying to say it's both for all platforms and also in some sense "no" platform. Maybe the term `shared library` doesn't fit well? I'm not sure that's a general term?

File: src/Templates/src/templates/maui-lib/Platforms/iOS/PlatformClass1.cs (hunk 1 above)
Comment (mattleibow) [reply src/Templates/src/templates/maui-lib/Platforms/iOS/PlatformClass1.cs:5]: No, there is a separate MacCatalyst folder for that.

Comment (Eilon) [inline src/Templates/src/templates/maui-lib/Platforms/iOS/PlatformClass1.cs:5]: Is this also used on MacCatalyst?

Comment (mattleibow) [reply src/Templates/src/templates/maui-lib/Platforms/iOS/PlatformClass1.cs:5]: no. There is a MacCat folder for that.

File: src/Templates/src/templates/maui-lib/Class1.cs (hunk 2 above)
Comment (mattleibow) [reply src/Templates/src/templates/maui-lib/Class1.cs]: I updated the description to say this is always included.

File: src/Templates/src/templates/maui-lib/.template.config/template.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{
+    "$schema": "http://json.schemastore.org/template",
+    "author": "Microsoft",
+    "classifications": [ "MAUI", "Android", "iOS", "macOS", "Mac Catalyst", "WinUI" ],
+    "identity": "Microsoft.Maui.MauiLib",
+    "groupIdentity": "Microsoft.Maui",
+    "name": ".NET MAUI Class Library",
+    "description": "A project for creating a .NET MAUI class library",
+    "shortName": "mauilib",
```
Comment (mattleibow) [inline src/Templates/src/templates/maui-lib/.template.config/template.json:9]: Are we happy with `mauilib`? `maui-lib`? `maui-classlib`?

Comment (Eilon) [reply src/Templates/src/templates/maui-lib/.template.config/template.json:9]: I prefer shorter. `mauilib`

File: src/Templates/src/templates/maui-lib/MauiLib1.csproj [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+	<PropertyGroup>
+		<TargetFrameworks>net6.0;net6.0-ios;net6.0-android;net6.0-maccatalyst</TargetFrameworks>
+		<TargetFrameworks Condition="'$(OS)' == 'Windows_NT'">$(TargetFrameworks);net6.0-windows10.0.19041</TargetFrameworks>
```
Comment (Eilon) [inline src/Templates/src/templates/maui-lib/MauiLib1.csproj:5]: Not sure what the plan is, but is this version of win10 correct? Or is it too high?

Comment (mattleibow) [reply src/Templates/src/templates/maui-lib/MauiLib1.csproj:5]: That is the TFM, but the minimum version set later will make it work on older OS.

**Review by Eilon — APPROVED** [review]

**Review by Eilon — CHANGES_REQUESTED** [review]
I was just talking to some ASP.NET Core folks and how their project templates will take advantage of new SDK and compiler features such as auto-imports instead of using's and also using the new `namespace Foo;` top-level namespace feature. I think the MAUI templates should match these new patterns as well so that the newest framework in .NET will also be showcasing the newest features.

@DamianEdwards - I believe you know the latest & greatest features that templates should be using, such as how to enable SDK default imports, etc.?

**Comment by DamianEdwards** [comment]
@Eilon yep, although there are some planned changes for the implicit usings feature in rc.1 that would change how the template opts-in, see dotnet/sdk#19521 for details.

For other modern things that we've adopted in the other templates that you might consider:
- file-scoped namespace declarations, i.e. `namespace MyGreatNamespace;`
- enabling nullable reference types by default, i.e. `<Nullable>enable</Nullable>` in the .csproj (and any code changes required)

**Review by Eilon — APPROVED** [review]
Signing off for now. We need to update all the project templates to the new pattern too anyway.

**Pull request merged**